File: C/Binary Tree - BFS/1161. Maximum Level Sum of a Binary Tree.c
/* 1161. Maximum Level Sum of a Binary Tree 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/maximum-level-sum-of-a-binary-tree/submissions/1385726317/ */

int find_height(struct TreeNode * root)
{
    if(root)
    {
        int l_height = find_height(root->left);
        int r_height = find_height(root->right);

        return (l_height >= r_height) ? (l_height + 1) : (r_height + 1);
    }
    return 0;
}

void traversal(struct TreeNode * root, int cur_level, int * level_arr_sum)
{
    if(root)
    {
        level_arr_sum[cur_level] += root->val;

        traversal(root->left, cur_level + 1, level_arr_sum);
        traversal(root->right, cur_level + 1, level_arr_sum);
    }
    return;
}

int maxLevelSum(struct TreeNode* root) 
{
    int max_val = INT_MIN;
    int smallest_level = -1;

    int height = find_height(root);

    int * level_arr = (int *)malloc(sizeof(int) * (height + 1)); 
    int i=0;
    for(i=0; i<height+1; i++)
    {
        level_arr[i] = 0;
    }

    traversal(root, 1, level_arr);  
    for(i=1; i<height+1; i++)
    {
        if(level_arr[i] > max_val)
        {
            max_val = level_arr[i];
            smallest_level = i;
        }
    }
    return smallest_level;  
}

File: C#/Binary Tree - BFS/1161. Maximum Level Sum of a Binary Tree.cs
/* 1161. Maximum Level Sum of a Binary Tree 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/maximum-level-sum-of-a-binary-tree/submissions/1385726063/ */

public class Solution {
    public int MaxLevelSum(TreeNode root)
    {
        var traverse = new Queue<TreeNode>();
        traverse.Enqueue(root);
        
        var maxLevel = 1;
        var maxLevelSum = root.val;

        var currentLevel = 0;

        while (traverse.Count != 0)
        {
            currentLevel++;
            var levelCount = traverse.Count;
            var currentLevelSum = 0;
            
            for (int i = 0; i < levelCount; i++)
            {
                var node = traverse.Dequeue();

                if (node.left is not null)
                {
                    traverse.Enqueue(node.left);
                }
                
                if (node.right is not null)
                {
                    traverse.Enqueue(node.right);
                }
                
                currentLevelSum += node.val;
            }

            if (maxLevelSum < currentLevelSum)
            {
                maxLevelSum = currentLevelSum;
                maxLevel = currentLevel;
            }
        }
        
        return maxLevel;
    }
}